### DISCUSSION TITLE
剥离元数据管理+集群协调到独立的服务

### DISCUSSION THREAD

**Discussion by lookhua** [apache/iotdb#4121] [Ideas]
目前集群模式的部署和维护在k8s部署的条件下显得非常困难，扩缩容的时候没有办法动态调整集群配置参数。我这边认为可以将集群的元数据管理+集群协调独立到单独的服务，这样集群节点 就可以任意的进行动态扩缩容，不知道大家认为这种架构设计是不是更云原生一些？

**Comment by jixuan1989** [comment]
目前设计上，分布式版本下一轮大迭代会把元数据管理集中到几台节点上，而不是全部节点。
进一步，可以考虑将元数据管理接口话，这样社区里就可以提供多种实现模式了，包括自实现，或者第三方服务。
